src/components/TodoItem/TodoItem.props.ts

```diff
@@ -0,0 +1,12 @@
+import { Todo } from '../../types/Todo';
+
+export type Props = {
+  todo: Todo;
+  onDelete: (todoId: number) => void;
+  selectedTodos: number[];
+  setSelectedTodos: (todoId: number[]) => void;
+  onUpdate: (todoId: number, data: Partial<Todo>) => void;
```

Comment:
```suggestion
  onUpdate: (todoId: number, data: RequireAtLeastOne<Todo, 'completed' | 'title'>) => void;
```

Are you sure that it's safe to make all Todo properties optional? Now you can pass even empty Todo. Better use another approach.